reference/OpenAPI.json

```diff
@@ -381,6 +381,1013 @@
+            "$ref": "#/components/schemas/DiscountUnitVouchersEffectTypes",
+            "description": "Defines how the unit is added to the customer's order."
+          },
+          "unit_type": {
+            "type": "string",
+            "description": "The product deemed as free, chosen from product inventory (e.g. time, items)."
+          },
+          "product": {
+            "$ref": "#/components/schemas/SimpleProductDiscountUnit",
+            "description": "Contains information about the product."
+          },
+          "sku": {
+            "$ref": "#/components/schemas/SimpleSkuDiscountUnit"
+          }
+        }
+      },
+      "DiscountUnitMultiple": {
+        "type": "object",
+        "title": "Unit Multiple",
+        "properties": {
+          "type": {
+            "type": "string",
+            "enum": [
+              "UNIT"
+            ],
+            "description": "Discount type."
+          },
+          "effect": {
+            "type": "string",
+            "enum": [
+              "ADD_MANY_ITEMS"
+            ],
+            "description": "Defines how the discount is applied to the customer's order."
+          },
+          "units": {
+            "type": "array",
+            "items": {
+              "$ref": "#/components/schemas/DiscountUnitMultipleOneUnit"
+            }
+          }
+        },
+        "required": [
+          "type",
+          "effect",
+          "units"
+        ]
+      },
+      "DiscountUnitMultipleOneUnit": {
+        "type": "object",
+        "title": "One Unit",
+        "properties": {
+          "type": {
+            "type": "string",
+            "enum": [
+              "UNIT"
+            ],
+            "description": "Discount type."
+          },
+          "unit_off": {
+            "type": "number",
+            "description": "Number of units to be granted a full value discount."
+          },
+          "unit_off_formula": {
+            "type": "string"
+          },
+          "effect": {
+            "type": "string",
+            "enum": [
+              "ADD_NEW_ITEMS",
+              "ADD_MISSING_ITEMS"
+            ],
+            "description": "Defines how the unit is added to the customer's order.\n\n"
+          },
+          "unit_type": {
+            "type": "string",
+            "enum": [
+              "UNIT"
+            ],
+            "description": "The product deemed as free, chosen from product inventory (e.g. time, items)."
+          },
+          "product": {
+            "$ref": "#/components/schemas/SimpleProductDiscountUnit",
+            "description": "Contains information about the product."
+          },
+          "sku": {
+            "$ref": "#/components/schemas/SimpleProductDiscountUnit",
+            "description": "Contains information about the sku."
+          }
+        },
+        "required": [
+          "type",
+          "unit_off",
+          "effect"
+        ]
+      },
+      "DiscountAmount": {
```

Comment:
The other types for `discount` in response (https://docs.voucherify.io/v2018-08-01-piotr-502/reference/validate-voucher) don't have required properties (expect the UNIT_MULTIPLE). I'm not sure if all the fields are required :)